gqcp/include/Basis/Transformations/SimpleOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,149 @@
+    using Scalar = _Scalar;
+
+    // The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+    using DerivedOrbitalRotationGenerators = _DerivedOrbitalRotationGenerators;
+
+private:
+    // The number of spatial orbitals that can be rotated using these orbital rotation generators.
+    size_t number_of_spatial_orbitals;
+
+    // The strict upper/lower triangle of the kappa matrix.
+    VectorX<Scalar> kappa_vector;
+
+
+public:
+    /**
+     *  MARK: Constructors
+     */
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa vector.
+     * 
+     *  @param  kappa_vector        The orbital rotation generators represented as a vector that corresponds to the strict upper/lower triangle of the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const VectorX<Scalar>& kappa_vector) :
+        kappa_vector {kappa_vector},
+        number_of_spatial_orbitals {strictTriangularRootOf(kappa_vector.size())} {}
+
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa matrix.
+     * 
+     *  @param  kappa_matrix        The orbital rotation generators represented as a vector that corresponds to the full antisymmetric the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const SquareMatrix<Scalar>& kappa_matrix) :
+        SimpleOrbitalRotationGenerators(kappa_matrix.pairWiseStrictReduced()) {}
+
+
+    /**
+     *  MARK: Named Constructors
+     */
+
+    /**
+     *  Construct orbital rotation generators by adding redundant (i.e. 0) occupied-virtual and virtual-virtual generators to the given occupied-occupied generators.
+     * 
+     *  @param o_o_generators       The occupied-occupied orbital rotation generators.
+     *  @param K                    The total number of spatial orbitals.
+     * 
+     *  @return The 'full' orbital rotation generators from the given occupied-occupied generators.
+     */
+    static DerivedOrbitalRotationGenerators FromOccOcc(const DerivedOrbitalRotationGenerators& o_o_generators, const size_t K) {
+        SquareMatrix<Scalar> kappa_full_matrix = SquareMatrix<Scalar>::Zero(K);
+
+        kappa_full_matrix.topLeftCorner(o_o_generators.numberOfSpatialOrbitals(), o_o_generators.numberOfSpatialOrbitals()) = o_o_generators.asMatrix();
+        return DerivedOrbitalRotationGenerators(kappa_full_matrix);
+    }
+
+
+    /**
+     *  MARK: Access
+     */
+
+    /**
+     *  @return The antisymmetric orbital rotation generator matrix kappa.
+     */
+    const SquareMatrix<Scalar> asMatrix() const {
+
+        const auto kappa_matrix = SquareMatrix<Scalar>::FromStrictTriangle(this->kappa_vector);  // Lower triangle only.
+        const SquareMatrix<Scalar> kappa_matrix_transpose = kappa_matrix.transpose();
+
+        // Add the antisymmetric component and return the matrix representation.
+        return kappa_matrix - kappa_matrix_transpose;
+    }
```

Comment:
Anti-symmetric or anti-Hermitian?

Reply:
Since we're enabling complex generators as well, anti-Hermitian! I'll adjust it.